tailer/fswatcher/linereader.go

```diff
@@ -41,19 +44,32 @@ func (r *lineReader) ReadLine(file io.Reader) (string, bool, error) {
 		err error
 		buf = make([]byte, 512)
 		n   = 0
+		reg = regexp.MustCompile(r.lineDelimiter)
```

Comment:
If the user has an invalid regular expression in `line_delimiter`, grok_exporter should fail on startup with a reasonable error message. Please add a check for this to `func (c *InputConfig) validate()` in `configV3.go`.

Reply:
If you compile the regular expression when loading the config, you might as well store the result so that you don't need to compile it again here.